scrape/helpers_test.go

```diff
@@ -46,25 +49,36 @@ func (a *collectResultAppender) AddFast(m labels.Labels, ref uint64, t int64, v
 	if err != nil {
 		return err
 	}
+	a.mtx.Lock()
 	a.result = append(a.result, sample{
 		metric: m,
 		t:      t,
 		v:      v,
 	})
+	a.mtx.Unlock()
 	return err
 }
 
 func (a *collectResultAppender) Add(m labels.Labels, t int64, v float64) (uint64, error) {
+	a.mtx.Lock()
 	a.result = append(a.result, sample{
 		metric: m,
 		t:      t,
 		v:      v,
 	})
+	a.mtx.Unlock()
 	if a.next == nil {
 		return 0, nil
 	}
 	return a.next.Add(m, t, v)
 }
 
-func (a *collectResultAppender) Commit() error   { return nil }
-func (a *collectResultAppender) Rollback() error { return nil }
+func (a *collectResultAppender) samples() []sample {
+	a.mtx.Lock()
```

Comment:
Should that be a read lock?

Reply:
I've used a normal mutex as the code is only used in tests so we have no particular performance concern.